code/game/machinery/ATMOSPHERICS/datum_pipeline.dm

```diff
@@ -71,21 +71,61 @@
 
 			if(result.len>0)
 				for(var/obj/machinery/atmospherics/pipe/item in result)
-					if(item.parent != src)
-						if(item.parent)
-							//Destroy the old pipeline so that the air is stored in the pipes
-							//This could be optimized significantly by making it merge item.parent into this pipeline instead (or vice versa) but I'm just fixing a bug here
-							qdel(item.parent)
-						members += item
-						possible_expansions += item
+					var/merged_other = FALSE
+					if(item.parent != src) // encountered a pipe that is not part of this pipeline
```

Comment:
```suggestion
					edge_check-- //REMOVE THE ONE AT THE BOTTOM, it's safe to move here I believe since nothing will exit the above for loop early
					if(item.parent == src) // encountered a pipe that is already a part of the pipeline
						continue
					var/merged_other = FALSE
					//everything after loses a tab
```